File: Python/rock-paper-sizzor.py
import random

# default player scores

player1_score = 0
player2_score = 0

# default game choices

choices = ["rock", "paper", "sizzor"]

# function to show game help

def game_help():
    print("Rock, Paper, Sizzor, Reset, Quit, Help")

# show the game help

game_help()

# start infinite game loop

running = True

while running:

    # let player1 choose

    player1_choice = str(input(f"\nPLAYER-1 [{player1_score}]--> "))

    # check player1 choice

    match (player1_choice):

        # check if player1 quits the game

        case "quit":
            print("\n--> Thanks for playing!")
            running = False
            continue

        # check if player1 asks for help

        case "help":
            game_help()
            continue

        # check if player1 asks for stats

        case "stats":
            print(f"\nPLAYER-1 [{player1_score}] vs PLAYER-2 [{player2_score}]")
            continue

        # check if player1 resets the game

        case "reset":
            player1_score = 0
            player2_score = 0
            continue

    # let player2 choose

    player2_choice = choices[random.randrange(2)]
    print(f"PLAYER-2 [{player2_score}]--> {player2_choice}\n")

    # check if player1 is the winner

    if (player1_choice == "rock" and player2_choice == "sizzor" or player1_choice == "paper" and player2_choice == "rock" or player1_choice == "sizzor" and player2_choice == "paper"):
        player1_score += 1
        print(f"PLAYER-1 [{player1_score}]--> Winner!")
        print(f"PLAYER-2 [{player2_score}]--> Looser!")

    # check if player2 is the winner

    elif (player2_choice == "rock" and player1_choice == "sizzor" or player2_choice == "paper" and player1_choice == "rock" or player2_choice == "sizzor" and player1_choice == "paper"):
        player2_score += 1
        print(f"PLAYER-1 [{player1_score}]--> Winner!")
        print(f"PLAYER-2 [{player2_score}]--> Looser!")

    # check if tie between players

    elif (player1_choice == player2_choice):
        print("--> Tie!")

    # check if invalid choice by player1

    else:
        print("--> Invalid choice!")

File: Lua/rock-paper-sizzor.lua
-- default run count

local run_count = 0

-- default player scores

local player1_score = 0
local player2_score = 0

-- default player choices

local player1_choice
local player2_choice

-- default game choices

local choices = {"rock", "paper", "sizzor"}

-- function to show game help

local function game_help()
    print("Rock, Paper, Sizzor, Stats, Reset, Quit, Help")
end

-- show the game help

game_help()

-- start infinite game loop

local running = true

while running do

    -- increment run count
    
    run_count = run_count + 1

    -- let player1 choose

    io.write(string.format("\n(%d) PLAYER-1 [%d]--> ", run_count, player1_score))
    player1_choice = io.read()

    -- check if player1 resets the game

    if player1_choice == "reset" then
        run_count = 0
        player1_score = 0
        player2_score = 0
        goto continue

    -- check if player1 asks for stats

    elseif player1_choice == "stats" then
        print(string.format("(%d) PLAYER-1 [%d] vs PLAYER-2 [%d]", run_count, player1_score, player2_score))

    -- check if player1 asks for help

    elseif player1_choice == "help" then
        game_help()
        goto continue

    -- check if player1 quits the game

    elseif player1_choice == "quit" then
        print("\n--> Thanks for playing!")
        running = false
        goto continue
    end

    -- let player2 choose

    player2_choice = choices[math.random(3)]
    print(string.format("(%d) PLAYER-2 [%d]--> %s\n", run_count, player2_score, player2_choice))

    -- check if player1 is the winner

    if player1_choice == "rock" and player2_choice == "sizzor" or player1_choice == "paper" and player2_choice == "rock" or player1_choice == "sizzor" and player2_choice == "paper" then
        player1_score = player1_score + 1
        print(string.format("(%d) PLAYER-1 [%d]--> Winner!", run_count, player1_score))
        print(string.format("(%d) PLAYER-2 [%d]--> Looser!", run_count, player2_score))

    -- check if player2 is the winner

    elseif player2_choice == "rock" and player1_choice == "sizzor" or player2_choice == "paper" and player1_choice == "rock" or player2_choice == "sizzor" and player1_choice == "paper" then
        player2_score = player2_score + 1
        print(string.format("(%d) PLAYER-2 [%d]--> Winner!", run_count, player2_score))
        print(string.format("(%d) PLAYER-1 [%d]--> Looser!", run_count, player1_score))

    -- check if tie between players

    elseif player1_choice == player2_choice then
        print("--> Tie!")

    -- check if invalid choice by player1

    else
        print("--> Invalid choice!")
    end

    ::continue::
end
